app/controllers/admin/problems_controller.rb

```diff
@@ -6,8 +6,18 @@ def new
     end
 
     def create
-      @contest = Contest.instance
-      @problem = @contest.problems.create problem_parameters
+      @contest    = Contest.instance
+      @problem    = @contest.problems.create problem_parameters
+      uploaded_io = params[:problem][:attachment]
+
+      if uploaded_io
+        attachment = @problem.attachments.create(original_filename: uploaded_io.original_filename,
```

Comment:
use build instead of create, so that DB saves only get triggered from a single point in the controller method